sql/parser/update.go

```diff
@@ -62,6 +64,15 @@ func (p *Parser) parseSetClause() (map[string]expr.Expr, error) {
 			return nil, newParseError(scanner.Tokstr(tok, lit), []string{"identifier"}, pos)
 		}
 
+		p.Unscan()
+		ref, err := p.parseFieldRef()
+		if err != nil {
+			return nil, newParseError(scanner.Tokstr(tok, lit), []string{"identifier"}, pos)
+		}
+		if len(ref) > 1 {
```

Comment:
Since we are dealing with `ValuePath`s now, I think the `parseSetClause` method should return something that takes that into account.
Otherwise, because of this `if`, we'd basically be doing this:

       Input                    Parsing                     Joining                  
     "foo.1.bar.0.1"  --> ["foo", "1', "bar", "0", "1"] --> "foo.1.bar.0.1"